CONTRIBUTING.md

```diff
@@ -71,6 +71,54 @@ $ git rebase master
 
 > Now your `feature/my-new-feature` branch is up-to-date with all the code in `upstream/master`.
 
+Running E2E Tests locally before commits
+----------------------------------------
+E2E Tests can be run against the Custom Login, Okta-Hosted Login and Resource servers
+
+Follow the steps below to run the tests locally:
+
+```bash
+# At project root
+composer install
+```
+To test the samples you will need the following configured in your developer org:
+
+* [A Web application](/okta-hosted-login#prerequisites)
+* [A SPA application](https://github.com/okta/samples-js-angular/tree/master/okta-hosted-login#prerequisites)
+* A test user account with a known username and password. Note that the USERNAME should be of the form "username@email.com"
+
+Once you have those resources setup, export their details as the following environment variables:
+
+```bash
+export ISSUER=https://{yourOktaDomain}.com/oauth2/default
+export CLIENT_ID={yourWebAppClientId}
+export CLIENT_SECRET={yourWebAppClientSecret}
+export SPA_CLIENT_ID={yourSpaAppClientId}
+export USERNAME={userName}
+export PASSWORD={password}
+```
+
+For Windows, please set the following environment variables:
+- `ISSUER`
+- `CLIENT_ID`
+- `CLIENT_SECRET`
+- `SPA_CLIENT_ID`
+- `USER_NAME`
+- `PASSWORD`
+
+> **NOTE:** Windows has USERNAME as a built-in system variable, hence set the USER_NAME environment variable for testing.
+
+Then run the E2E tests:
+
+```bash
+npm test
+```
+
+> **NOTE:** If you want to execute individual tests such as `npm run test:okta-hosted-login`, you will need to update the environment by running the following node script first:
+
+```bash
+node scripts/setup-env.js
```

Comment:
Can all this be done directly in the composer.json file instead of having to run more things?   

composer scripts

```
  'test:okta-hosted-login': "node scripts/setup-env.js && ......"
```

Reply:
Actually it's already being done in the `composer.json` scripts. Let me update the instructions.